src/init-conf.ts

```diff
@@ -0,0 +1,151 @@
+import { N9Log } from '@neo9/n9-node-log';
+import { N9Error } from '@neo9/n9-node-utils';
+import { classToPlain, plainToClass } from 'class-transformer';
+import { validate, ValidationError } from 'class-validator';
+
+import * as N9NodeRouting from './models/routing';
+import ConfValidationOptions = N9NodeRouting.ConfValidationOptions;
+
+function formatWhitelistErrors(
+	validationErrors: ValidationError[],
+	prefix: string,
+): [key: string, message: string][] {
+	const formattedWhitelistErrors: [key: string, message: string][] = [];
+	for (const validationError of validationErrors) {
+		const computedPrefix = prefix
+			? `${prefix}.${validationError.property}`
+			: validationError.property;
+		if (validationError.constraints?.whitelistValidation) {
+			formattedWhitelistErrors.push([
+				computedPrefix,
+				validationError.constraints.whitelistValidation,
+			]);
+		}
+		if (validationError.children) {
+			formattedWhitelistErrors.push(
+				...formatWhitelistErrors(validationError.children, computedPrefix),
+			);
+		}
+	}
+	return formattedWhitelistErrors;
+}
+
+function formatValidationErrors(
+	validationErrors: ValidationError[],
+	prefix: string,
+): [key: string, message: string[]][] {
+	const formattedErrors: [key: string, message: string[]][] = [];
+	for (const validationError of validationErrors) {
+		const computedPrefix = prefix
+			? `${prefix}.${validationError.property}`
+			: validationError.property;
+		if (validationError.constraints) {
+			const formattedError: [key: string, message: string[]] = [computedPrefix, []];
+			for (const key of Object.keys(validationError.constraints)) {
+				formattedError[1].push(`${key}: ${validationError.constraints[key]}`);
+			}
+			formattedErrors.push(formattedError);
+		}
+		if (validationError.children) {
+			formattedErrors.push(...formatValidationErrors(validationError.children, computedPrefix));
+		}
+	}
+	return formattedErrors;
+}
+
+function assertClassTypeIsGiven(validationOptions: ConfValidationOptions): void {
+	if (!validationOptions.classType) {
+		throw new N9Error(
+			'N9NodeRouting configuration validation options are not correct validation is enabled but' +
+				' no classType is given',
+			500,
+			validationOptions,
+		);
+	}
+}
+
+function handleValidationErrors(
+	validationErrors: ValidationError[],
+	logger: N9Log,
+	options: ConfValidationOptions,
+): void {
+	const noError = validationErrors.length === 0;
+	if (noError) return;
+
+	const validationErrorsWithoutExcludeProperties = classToPlain(validationErrors);
+	if (!options.formatValidationErrors) {
+		logger.error('Configuration is not valid', {
+			validationErrors: validationErrorsWithoutExcludeProperties,
```

Comment:
Attention à ne pas mettre un objet complexe en content de logger. 

```suggestion
			argString: JSON.stringify(validationErrorsWithoutExcludeProperties),
```

Idem pour les autres usages

Reply:
Je ne comprends pas ce que ça change. Peux tu m'expliquer ?
Je préfère le système de logger décider de ce qu'il veut faire. En plus, c'est le résultat d'un classToPlain, l'objet est déjà un plain donc moins complexe.

Reply:
J'ai ajouté le stringify. Ne pas oublier de faire ça dans le logger pour la suite et non ici.
Done